integration-tests/testconfig/testconfig.go

```diff
@@ -0,0 +1,551 @@
+package testconfig
+
+import (
+	"encoding/base64"
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/pelletier/go-toml/v2"
+	"github.com/pkg/errors"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+
+	ctf_config "github.com/smartcontractkit/chainlink-testing-framework/config"
+	ctf_test_env "github.com/smartcontractkit/chainlink-testing-framework/docker/test_env"
+	"github.com/smartcontractkit/chainlink-testing-framework/logging"
+	"github.com/smartcontractkit/chainlink-testing-framework/utils/osutil"
+	a_config "github.com/smartcontractkit/chainlink/integration-tests/testconfig/automation"
+	f_config "github.com/smartcontractkit/chainlink/integration-tests/testconfig/functions"
+	keeper_config "github.com/smartcontractkit/chainlink/integration-tests/testconfig/keeper"
+	lp_config "github.com/smartcontractkit/chainlink/integration-tests/testconfig/logpoller"
+	ocr_config "github.com/smartcontractkit/chainlink/integration-tests/testconfig/ocr"
+	vrf_config "github.com/smartcontractkit/chainlink/integration-tests/testconfig/vrf"
+	vrfv2_config "github.com/smartcontractkit/chainlink/integration-tests/testconfig/vrfv2"
+	vrfv2plus_config "github.com/smartcontractkit/chainlink/integration-tests/testconfig/vrfv2plus"
+)
+
+type TestConfig struct {
+	ChainlinkImage         *ctf_config.ChainlinkImageConfig `toml:"ChainlinkImage"`
```

Comment:
@Tofel could we have `[]ChainlinkImage`? This way we could set different versions of core for nodes for compatibility testing https://smartcontract-it.atlassian.net/browse/TT-664 and https://smartcontract-it.atlassian.net/browse/TT-665

Reply:
We could, but why not add that in one of the tickets mentioned?